repos/system_upgrade/el7toel8/libraries/rpms.py

```diff
@@ -30,24 +30,27 @@ def create_lookup(model, field, key, context=stdlib.api):
     """
     data = getattr(next((m for m in context.consume(model)), model()), field)
     try:
-        return {getattr(obj, key) for obj in data} if data else {}
+        return {tuple(getattr(obj, key) for key in keys) for obj in data} if data else {}
```

Comment:
I know this has been before, but this generates a set if there's data otherwise a dictionary. Quite strange, don't you think? Maybe you can fix it here by changing the `else {} `to `else set()`

Reply:
Sure, thanks a lot! I have missed that twice

Reply:
Should be fixed and covered with tests